src/Simplex/Messaging/Agent/Transmission.hs

```diff
@@ -91,13 +90,50 @@ data ACommand (p :: AParty) where
 
 deriving instance Show (ACommand p)
 
+type Message = ByteString
+
 data AMessage where
   HELLO :: VerificationKey -> AckMode -> AMessage
   REPLY :: SMPQueueInfo -> AMessage
   A_MSG :: MsgBody -> AMessage
 
--- A_ACK :: AgentMsgId -> AckStatus -> AMessage
--- A_DEL :: AMessage
+parseMessage :: Message -> Either ErrorType AMessage
+parseMessage msg = case B.words msg of
+  -- ["HELLO", key, ackMode] -> Right $ HELLO key $ ackMode am
```

Comment:
This should be easily fixable with `HELLO key <$> ackMode am` (ackMode am returns Either and `HELLO key` is a function that takes one parameter, so it can be used with fmap).